src/libraries/System.Diagnostics.Debug/tests/System.Diagnostics.Debug.Tests.csproj

```diff
@@ -26,4 +18,13 @@
     <Compile Include="EmptyAttributeTests.cs" />
     <Compile Include="XunitAssemblyAttributes.cs" />
   </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="$(CoreLibProject)" />
+    <ProjectReference Include="$(LibrariesProjectRoot)System.Runtime\src\System.Runtime.csproj;
+                               $(LibrariesProjectRoot)System.Threading\src\System.Threading.csproj"
+                      SkipUseReferenceAssembly="true" />
+    <!-- Do not reference since we build part of the source code for tests. -->
+    <ProjectRefence Include="..\src\System.Diagnostics.Debug.csproj" ReferenceOutputAssembly="false" />
```

Comment:
Did you consider making the default be `ReferenceOutputAssembly` = false for anything in the shared framework?  It seems like we're making these tests be a slightly more complicated by getting the references from two different mechanisms.  If the goal is just to aide in the inner-loop (single command for rebuilding and testing) then why not have the P2P only build?

Reply:
In this specific case that you linked to I'm actually making the test simpler as before we required setting `DefaultReferenceExclusion` which isn't needed anymore as if setting `ReferenceOutputAssembly` to false, it automatically filters the targeting pack reference out.

Regarding your other comment, I guess we could make it the default but I wonder if the difference matters:
a) we use the P2P as is and reference its output assembly and filter the assembly that comes via the targeting pack out.
b) we use the P2P without referencing the output assembly (intentionally setting `ReferenceOutputAssembly` to false) and don't filter the assembly that comes via the targeting pack out.

Don't have a strong opinion here so I'm ok with both.

Reply:
> In this specific case that you linked

Yeah, I realized this case was different but needed to choose a line to comment on so we could have a conversation thread :)

I would prefer that the default be `ReferenceOutputAssembly='false'` for things in the shared framework so that we keep the ref-pack as the source of the references.  It feels like we're closer to the real product this way and the src-reference is just a build convenience when building from the command-line.  As you mention `DefaultReferenceExclusion` is still needed, we'd also still need `SkipUseReferenceAssembly` as well.

Reply:
Thanks for the feedback which makes sense. I just changed this.